### ISSUE TITLE
Bug Report: Game crashes when a score is larger than 9,999,999 (seven digits).

### ISSUE THREAD

**Issue by Shaunyman** [FunkinCrew/Funkin#3631]
### Issue Checklist

- [X] I have properly named my issue
- [X] I have checked the Issues/Discussions pages to see if my issue has already been reported

### Platform

Itch.io (Downloadable Build) - Windows

### Browser

None

### Version

0.4.1

### Description (include any images, videos, errors, or crash logs)

Note: This issue has not been tested in v0.5.0 or v0.5.1. However, I believe the issue will still occur as the freeplay score only reflects 7 digits in all current versions of the game (v0.5.1 being the newest). 

This is related to my last issue [https://github.com/FunkinCrew/Funkin/issues/3070](url) where I made a song super long that it broke the chart editor. The temporary issue there was to lower the audio bitrate from 41kHz (41,000 Hz), to something lower until the chart editor would work. I ended up getting the full 192 minute chart to work. The new issue that arises from this is that the score that I accumulated was 13,000,000 and change. I believe the issue is that the game's score at the top right of the freeplay menu is only 7 digits and my actual score is 8 digits, which is too many for the freeplay menu to show. The results screen however, can represent a score that is 10 digits (9,999,999,999 max). I believe that the crash arises when the game tries to show the results of a score being more than 7 digits. I have attached pictures of the error messages as well as a video of the crash occurring. Let me know if there's anything I can do to help!
![Scoreboard ok 3](https://github.com/user-attachments/assets/0d5a6b12-d952-41bd-91ed-e442d803f8c5)
![Scoreboard ok error 2](https://github.com/user-attachments/assets/2a7a33fc-c565-4817-a3b6-a1f3a64fd25c)
![Scoreboard ok error](https://github.com/user-attachments/assets/842287d7-2d22-410c-8fd6-d946432e7cbf)
![Scoreboard ok](https://github.com/user-attachments/assets/3f8af834-ad28-46ed-8a29-f035b1e3dc63)

https://github.com/user-attachments/assets/da7a709d-bf4e-4acf-9c67-e358007c00e1

### Steps to Reproduce

In order for this issue to be recreated, you must have a song that will produce a score of over 9,999,999 when played. For simplicity, this link is to the chart that I created and used to discover this issue: [https://gamebanana.com/mods/537502](url). You'll need to download the executable file in order to get a score for the song (WARNING: Completion of the mod chart will overwrite the save score for the tutorial on easy difficulty as the chart of the downloaded desktop version of the game! If you do not want your game to crash every time you go to the tutorial on easy after completion of this chart, do not download! This issue arises because when I was making the chart metadata, I copied the info in the tutorial section of the files because I couldn't figure out how to get my own song to show up in the freeplay menu and work as a normal song does). Simply finishing any song with a score of 8 digits or more will cause the game to crash when attempting to go back to the freeplay menu to see the score/play it again. I know this was probably confusing, so, if I can help clarify anything in any way, please let me know. Also, it would be a lot easier to explain the issue while in a discord call so that I can share my screen and explain it verbally. Let me know if there's anything I can do to help!

**Comment by NotHyper-474** [comment]
Bro's really out here passing the limits of the FNF engine

**Comment by AbnormalPoof** [comment]
Reproduced on macOS.

https://github.com/user-attachments/assets/77778d9a-2116-434a-9983-187308cdca11

**Comment by AbnormalPoof** [comment]
This bug also bricks the Freeplay menu in the sense that you can't highlight the song anymore without it crashing from the extremely high score.

**Comment by Raltyro** [comment]
the fact that result screen score digit is higher than freeplay score digit is insane

**Comment by Shaunyman** [comment]
> This bug also bricks the Freeplay menu in the sense that you can't highlight the song anymore without the game crashing from the extremely high score.

Yes! That's one of the issues I was trying to describe. Also, I think it's hilarious that you used my chart/issue to test something for someone else's issue XD

**Comment by Shaunyman** [comment]
> the fact that result screen score digit is higher than freeplay score digit is insane

It's a very long chart that will only get longer as they game continues to release new songs because those songs and all of their difficulty variants will also be added to the mod chart. It's gonna be like 8 hours long once the full game releases!

**Comment by AbnormalPoof** [comment]
Maxing out the score in the Results Screen appears to not make it crash.
![screenshot-2024-10-12-12-31-15](https://github.com/user-attachments/assets/fc5df5a3-8245-4c16-b555-069f150903af)

**Comment by AbnormalPoof** [comment]
The Story Menu also doesn't crash from the extremely high score.
![screenshot-2024-10-12-12-39-01](https://github.com/user-attachments/assets/e4043c7b-a65c-4a91-b2ca-2b84c83e00a9)

**Comment by Shaunyman** [comment]
> The Story Menu also doesn't crash from the extremely high score. ![screenshot-2024-10-12-12-39-01](https://private-user-images.githubusercontent.com/86753001/375975527-e4043c7b-a65c-4a91-b2ca-2b84c83e00a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg3OTUwMjksIm5iZiI6MTcyODc5NDcyOSwicGF0aCI6Ii84Njc1MzAwMS8zNzU5NzU1MjctZTQwNDNjN2ItYTY1Yy00YTkxLWIyY2EtMmI4NGM4M2UwMGE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEzVDA0NDUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhNGI3MzM1Mjg2NGI5ODYwNDMwZmFmZTlmN2RiNDIzNTUxYmUyZWEwNjhjMDU5Mjg3YTJiZWRhODkxZWE3OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LJJXcelriyCvGIboAaqPEGeuyGgPDq4WVuAy0t3c9cY)

So it only crashes in the freeplay menu and not the story mode menu? That's interesting that the freeplay menu is capped to 7 digits, but not the story mode menu. Although, I guess it does make sense since the freeplay menu actually has a certain number of digit slots designed into it and the story mode menu doesn't.

**Comment by Lasercar** [comment]
This issue has pending pull request.
